rpcserver.go

```diff
@@ -3233,16 +3233,22 @@ func (r *rpcServer) QueryRoutes(ctx context.Context,
 	routeResp := &lnrpc.QueryRoutesResponse{
 		Routes: make([]*lnrpc.Route, 0, in.NumRoutes),
 	}
+	
+	currentBlockHeight, err:=r.server.chanRouter.CurrentBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+	
```

Comment:
It could be that this block height isn't the same block height anymore that was used in FindRoutes, leading to a failure when the payment is executed.